packages/forms/src/Components/Select.php

```
@@ -1415,7 +1415,14 @@ public function getInValidationRuleValues(): ?array
             return array_keys($this->getOptionLabels(withDefaults: false));
         }
 
-        return blank($this->getOptionLabel(withDefault: false)) ? [] : null;
+        $optionLabel = $this->getOptionLabel(withDefault: false);
+
+        if ($this->isSearchable() && ! $optionLabel) {
```

I don't think we can trust that an absence of the option label indicates an absence of `getOptionLabelUsing()`, it could be that the option is invalid and no label is returned. In which case, if its also searchable, this would mean the validation is skipped, I think.

To be honest, I think that we should probably alter `getOptionLabel()` and `getOptionLabels()` to throw an exception if `$withDefault` is `false` and `$this->getOptionLabelUsing` is null, instead of silently failing. Users should be forced to implement the correct callbacks otherwise validation is not possible.

---

Hey @danharrin, I tested that, but the `setUp()` method already provides a default `getOptionLabelUsing()` and `getOptionLabelsUsing()` callbacks, so the `$this->getOptionLabelUsing` is never `null`, so the exception is never thrown. Could work around it by detecting a custom option label set using e.g. a boolean property, but that would become a bit dirty quickly I think. What do you think?